src/main.rs

```diff
@@ -1970,12 +2048,57 @@ fn cmd_check(
             // more clever comparisons.
             if store.config.exemptions != pruned_exemptions {
                 warn!("Your supply-chain has unnecessary exemptions which could be relaxed or pruned.");
-                warn!("Consider running `cargo vet prune` to prune unnecessary exemptions and imports.");
+                warn!("  Consider running `cargo vet prune` to prune unnecessary exemptions and imports.");
             } else if store.imports != pruned_imports {
                 warn!("Your supply-chain has unnecessary imports which could be pruned.");
-                warn!("Consider running `cargo vet prune` to prune unnecessary imports.");
+                warn!("  Consider running `cargo vet prune` to prune unnecessary imports.");
             }
         }
+
+        // Check for wildcard audits which will be expiring soon, and warn about them.
+        {
+            let today = <chrono::DateTime<chrono::Utc>>::from(SystemTime::now()).date_naive();
+            let expire_date = today + *WILDCARD_AUDIT_EXPIRATION_DURATION;
+
+            let mut expired = Vec::new();
+            let mut expiring_soon = Vec::new();
+            for (name, audits) in &store.audits.wildcard_audits {
+                // Check whether there are any audits expiring by the expiration date. Of those
+                // audits, check whether all of them are already expired (to change the warning
+                // message to be more informative).
+                let mut any = false;
+                let mut is_expired = true;
+                for entry in audits
+                    .iter()
+                    .filter(|e| e.should_suggest_renewal(expire_date))
+                {
+                    any = true;
+                    is_expired &= entry.should_suggest_renewal(today);
+                }
+                if any {
+                    if is_expired {
+                        expired.push(name);
+                    } else {
+                        expiring_soon.push(name);
+                    }
+                }
+            }
+
+            if !(expired.is_empty() && expiring_soon.is_empty()) {
+                if !expiring_soon.is_empty() {
+                    let expiring = errors::FormatShortList::new(expiring_soon);
+                    warn!("Your audit set contains wildcard audits for {expiring} which expire within the next {WILDCARD_AUDIT_EXPIRATION_STRING}.");
+                }
+                if !expired.is_empty() {
+                    let expired = errors::FormatShortList::new(expired);
+                    warn!(
+                        "Your audit set contains wildcard audits for {expired} which have expired."
+                    );
+                }
+                warn!("  Consider running `cargo vet renew --expiring` or adding `suggest-renewal = false` to the wildcard entries in audits.toml.");
+            }
```

Comment:
It'd be nice to pull this out into a separate function so that we can have some unit tests just on this logic.